deals/deals_test.go

```diff
@@ -24,9 +25,43 @@ func TestMain(m *testing.M) {
 }
 
 func TestStore(t *testing.T) {
-	numMiners := []int{1}
+	numMiners := []int{1, 3, 5}
 	for _, nm := range numMiners {
-		t.Run(fmt.Sprintf("CantMiners%d", nm), storeMultiMiner(t, nm))
+		t.Run(fmt.Sprintf("CantMiners%d", nm), func(t *testing.T) {
+			dnet, _, _, close := tests.CreateLocalDevnet(t, nm)
+			defer close()
+			m, err := New(dnet.Client, WithImportPath(filepath.Join(os.TempDir(), "imports")))
+			checkErr(t, err)
+			_, err = storeMultiMiner(m, dnet, nm, randomBytes(1000))
+			checkErr(t, err)
+		})
+	}
+}
+
+func TestRetrieve(t *testing.T) {
```

Comment:
A new test that tests whole cycle of storing something, retriving it, and checking all things went as expected.